### ISSUE TITLE
Customer Requisitions: order types with different max. orders per period are working incorrectly

### ISSUE THREAD

**Issue by chastichno** [openmsupply/mobile#3180]
## Describe the bug

When creating a new customer requisition the value of "max. orders per period" is being set once user chose the order type. When user changes order type the list of available periods are set for previously chosen order type.

### To reproduce

Create a program with multiple order types. One of the type must have "max. orders per period" set to zero (or must have no orders left in this period)
1. Go to 'Customer Requisitions' -> 'New Requisition'
2. Select the customer and program.
3. Select an order type with "max. orders per period"=0
4. You should see that no periods are available for this order type.
5. Change selected order type to the type with "max. orders per period">0 (that has orders left for this period)
6. See error. "No periods available".

It is also working in the opposite way. Do steps 1. and 2.

3. Select an order type with "max. orders per period">0
4. You should see that you can select a period (but don't select it!)
5. Change selected order type to the type with "max. orders per period"=0
6. See error. Now you can select a period for the order type with no orders left.

### Expected behaviour

After selecting another period the list of available periods should refresh.

### Version and device info

- App version: 6.0.0-rc7

**Comment by Mai21** [comment]
Related to this issue, should it be "No order type available" when "max. orders per period"=0 and there is no other order types available? 

Detail here.
1. There is only one order type with the program
2. Created requisitions until max orders in this period
3. When the program selected after step 2,  we can still choose the order type. After selecting this, "No periods available" shows. 

I was not sure I better create a new issue or not...

**Comment by joshxg** [comment]
Closed by #3184
